realm/realm-library/src/main/java/io/realm/RealmQuery.java

```diff
@@ -82,6 +84,15 @@
      * to run it.
      */
     static <E extends RealmModel> RealmQuery<E> createQuery(Realm realm, Class<E> clazz) {
+        // Warn on query being executed on UI thread if isAllowQueriesOnUiThread is set to true, throw otherwise
+        if (realm.getSharedRealm().capabilities.isMainThread()) {
+            if (realm.getConfiguration().isAllowQueriesOnUiThread()) {
+                RealmLog.warn("It is not recommended to run queries on the UI thread as it may lead to a drop of frames or ANRs. Please consider doing so from another thread instead.");
+            } else {
+                throw new RealmException("There exists an opt-out for running queries on the UI thread. By default Realm allows queries from the UI thread. You can alternatively opt in by using 'RealmConfiguration.allowQueriesOnUiThread'.");
```

Comment:
I guess we don't explicitly document throwing RealmExceptions from all the queries, but maybe some class level documentation about this pointing to the async variants?